style/properties/shorthands/text.mako.rs

```diff
@@ -69,21 +65,26 @@
         #[allow(unused)]
         fn to_css<W>(&self, dest: &mut CssWriter<W>) -> fmt::Result where W: fmt::Write {
             use crate::values::specified::TextDecorationLine;
+            use crate::values::specified::Color;
 
-            let (is_solid_style, is_current_color, is_auto_thickness) =
+            let (is_solid_style, is_current_color) =
             (
-            % if engine == "gecko":
                 *self.text_decoration_style == text_decoration_style::SpecifiedValue::Solid,
-                *self.text_decoration_color == specified::Color::CurrentColor,
-                self.text_decoration_thickness.is_auto()
+                *self.text_decoration_color == Color::CurrentColor
+            );
+
+            % if engine == "gecko":
+                let is_auto_thickness = self.text_decoration_thickness.is_auto();
             % else:
-                true, true, true
+                let is_auto_thickness = true;
             % endif
-            );
 
             let mut has_value = false;
             let is_none = *self.text_decoration_line == TextDecorationLine::none();
             if (is_solid_style && is_current_color && is_auto_thickness) || !is_none {
+                if has_value {
+                    dest.write_char(' ')?;
+                }
```

Comment:
Why are you adding this? `has_value` is false, and even if it could be true you shouldn't change the logic.

Reply:
I will change it!!